Grid/qcd/action/gauge/LLRGaugeAction.h

```diff
@@ -0,0 +1,162 @@
+//
+// Created by dc-bonn2 on 9/17/25.
+//
+
+#include <Grid/qcd/llr_hmc/llr_hmc.h>
+
+#ifndef GRID_TELOS_DEVEL_LLRGAUGEACTION_H
+#define GRID_TELOS_DEVEL_LLRGAUGEACTION_H
+
+NAMESPACE_BEGIN(Grid);
+
+//template<class Gimpl>
+template<typename GaugeAction, class Gimpl>
+class LLRGaugeAction  : public Action<typename Gimpl::GaugeField> {
+public:
+
+    INHERIT_GIMPL_TYPES(Gimpl);
+
+    using Action<GaugeField>::S;
+    using Action<GaugeField>::Sinitial;
+    using Action<GaugeField>::deriv;
+    using Action<GaugeField>::refresh;
+    using Action<GaugeField>::set_struct_llrparams;
+    using Action<GaugeField>::get_struct_llrparams;
+
+    // instantiating the llr_hmc helper class via the namespace LLR
+    namespace_LLR::llrparams* p_llrparams_s;
+    namespace_LLR::llr_hmc* p_llr_hmc_o = new namespace_LLR::llr_hmc();
+    /////////////////////////////////////////////////////////////
+    /// [Constructors]
+    /////////////////////////////////////////////////////////////
+    explicit LLRGaugeAction(RealD beta_):
+                beta(beta_){};
+
+    explicit LLRGaugeAction(namespace_LLR::llrparams* llrparams_, RealD beta_):
+                beta(beta_),
+                p_llrparams_s(llrparams_){};
+
+    virtual std::string action_name() {return "LLRGaugeAction";}
+    /////////////////////////////////////////////////////////////
+    /// [Logger]
+    /////////////////////////////////////////////////////////////
+    virtual std::string LogParameters(){
+        std::stringstream sstream;
+        sstream << GridLogMessage << "[LLRGaugeAction] Beta: " << beta << std::endl;
+        return sstream.str();
+    }
+    /////////////////////////////////////////////////////////////
+    /// [Actions]
+    /////////////////////////////////////////////////////////////
+    virtual RealD GActionS(const GaugeAction &p_GaugeAction, const GaugeField &U) {
+        RealD gauge_action = 0.0;
+        //gauge_action =  p_GaugeAction.S(U);
+        std::cout << GridLogLLR << B_GREEN << "Gauge action LLRGaugeAction      ----->: "<< gauge_action << C_RESET <<std::endl;
+        return gauge_action;
+    };
+
+    virtual void refresh(const GaugeField &U, GridSerialRNG &sRNG, GridParallelRNG &pRNG){};
+
+    virtual RealD S(const GaugeField &U) {
+        RealD plaq = WilsonLoops<Gimpl>::avgPlaquette(U);
+        RealD vol = U.Grid()->gSites();
+        //RealD action = beta * (1.0 - plaq) * (Nd * (Nd - 1.0)) * vol * 0.5;
+        RealD action = (1.0 - plaq) * (Nd * (Nd - 1.0)) * vol * 0.5;
+        std::cout << GridLogLLR << B_GREEN << "plaq(nb) LLRGaugeAction         ----->: "<< plaq   << C_RESET <<std::endl;
+        std::cout << GridLogLLR << B_GREEN << "Vol(nb)  LLRGaugeAction         ----->: "<< vol    << C_RESET <<std::endl;
+        std::cout << GridLogLLR << B_GREEN << "S(nb)    LLRGaugeAction         ----->: "<< action << C_RESET <<std::endl;
```

Comment:
Now that we ostensibly have the code working, this level of logging is absurdly verbose; can it be removed?